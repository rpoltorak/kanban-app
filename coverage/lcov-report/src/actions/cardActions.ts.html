<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/actions/cardActions.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/actions</a> cardActions.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { Action } from "redux";
import uuid from "uuid";
&nbsp;
import CardModel from "../models/cardModel";
&nbsp;
export enum CardActionTypes {
  CARD_CREATE = "CARD_CREATE",
  CARD_DELETE = "CARD_DELETE",
  CARD_UPDATE = "CARD_UPDATE",
}
&nbsp;
export type CardAction = CreateCardAction | DeleteCardAction | UpdateCardAction;
&nbsp;
export interface CreateCardAction extends Action {
  type: CardActionTypes.CARD_CREATE;
  payload: {
    card: CardModel;
  };
}
&nbsp;
export interface DeleteCardAction extends Action {
  type: CardActionTypes.CARD_DELETE;
  payload: {
    cardId: string;
    laneId: string;
  };
}
&nbsp;
export interface UpdateCardAction extends Action {
  type: CardActionTypes.CARD_UPDATE;
  payload: {
    cardId: string;
    changes: Partial&lt;CardModel&gt;;
  };
}
&nbsp;
export function createCard(title: string, description: string, laneId: string): CreateCardAction {
  return {
    type: CardActionTypes.CARD_CREATE,
    payload: {
      card: {
        id: uuid.v4(),
        laneId,
        title,
        description,
      },
    },
  };
}
&nbsp;
export function deleteCard(cardId: string, laneId: string): DeleteCardAction {
  return {
    type: CardActionTypes.CARD_DELETE,
    payload: {
      cardId,
      laneId,
    },
  };
}
&nbsp;
export function updateCard(cardId: string, changes: Partial&lt;CardModel&gt;): UpdateCardAction {
  return {
    type: CardActionTypes.CARD_UPDATE,
    payload: {
      cardId,
      changes,
    },
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 23:33:21 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
