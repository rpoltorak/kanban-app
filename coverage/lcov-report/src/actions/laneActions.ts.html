<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/actions/laneActions.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/actions</a> laneActions.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { Action } from "redux";
import uuid from "uuid";
&nbsp;
import LaneModel from "../models/laneModel";
&nbsp;
export enum LaneActionTypes {
  LANE_CREATE = "LANE_CREATE",
  LANE_DELETE = "LANE_DELETE",
  LANE_UPDATE = "LANE_UPDATE",
}
&nbsp;
export type LaneAction = CreateLaneAction | DeleteLaneAction | UpdateLaneAction;
&nbsp;
export interface CreateLaneAction extends Action {
  type: LaneActionTypes.LANE_CREATE;
  payload: {
    lane: LaneModel,
  };
}
&nbsp;
export interface DeleteLaneAction extends Action {
  type: LaneActionTypes.LANE_DELETE;
  payload: {
    laneId: string,
  };
}
&nbsp;
export interface UpdateLaneAction extends Action {
  type: LaneActionTypes.LANE_UPDATE;
  payload: {
    laneId: string;
    changes: Partial&lt;LaneModel&gt;,
  };
}
&nbsp;
export function createLane(title: string): CreateLaneAction {
  return {
    type: LaneActionTypes.LANE_CREATE,
    payload: {
      lane: {
        id: uuid.v4(),
        title,
        cards: [],
      },
    },
  };
}
&nbsp;
export function deleteLane(laneId: string): DeleteLaneAction {
  return {
    type: LaneActionTypes.LANE_DELETE,
    payload: {
      laneId,
    },
  };
}
&nbsp;
export function updateLane(laneId: string, changes: Partial&lt;LaneModel&gt;): UpdateLaneAction {
  return {
    type: LaneActionTypes.LANE_UPDATE,
    payload: {
      laneId,
      changes,
    },
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 23:33:21 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
