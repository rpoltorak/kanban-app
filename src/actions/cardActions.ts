import { Action } from "redux";
import uuid from "uuid";

import CardModel from "../models/cardModel";

export enum CardActionTypes {
  CARD_CREATE = "CARD_CREATE",
  CARD_DELETE = "CARD_DELETE",
  CARD_UPDATE = "CARD_UPDATE",
}

export type CardAction = CreateCardAction | DeleteCardAction | UpdateCardAction;

export interface CreateCardAction extends Action {
  type: CardActionTypes.CARD_CREATE;
  payload: {
    card: CardModel;
  };
}

export interface DeleteCardAction extends Action {
  type: CardActionTypes.CARD_DELETE;
  payload: {
    cardId: string;
    laneId: string;
  };
}

export interface UpdateCardAction extends Action {
  type: CardActionTypes.CARD_UPDATE;
  payload: {
    cardId: string;
    changes: Partial<CardModel>;
  };
}

export function createCard(title: string, description: string, laneId: string): CreateCardAction {
  return {
    type: CardActionTypes.CARD_CREATE,
    payload: {
      card: {
        id: uuid.v4(),
        laneId,
        title,
        description,
      },
    },
  };
}

export function deleteCard(cardId: string, laneId: string): DeleteCardAction {
  return {
    type: CardActionTypes.CARD_DELETE,
    payload: {
      cardId,
      laneId,
    },
  };
}

export function updateCard(cardId: string, changes: Partial<CardModel>): UpdateCardAction {
  return {
    type: CardActionTypes.CARD_UPDATE,
    payload: {
      cardId,
      changes,
    },
  };
}
