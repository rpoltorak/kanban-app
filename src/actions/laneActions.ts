import { Action } from "redux";
import uuid from "uuid";

import LaneModel from "../models/laneModel";

export enum LaneActionTypes {
  LANE_CREATE = "LANE_CREATE",
  LANE_DELETE = "LANE_DELETE",
  LANE_UPDATE = "LANE_UPDATE",
}

export type LaneAction = CreateLaneAction | DeleteLaneAction | UpdateLaneAction;

export interface CreateLaneAction extends Action {
  type: LaneActionTypes.LANE_CREATE;
  payload: {
    lane: LaneModel,
  };
}

export interface DeleteLaneAction extends Action {
  type: LaneActionTypes.LANE_DELETE;
  payload: {
    laneId: string,
  };
}

export interface UpdateLaneAction extends Action {
  type: LaneActionTypes.LANE_UPDATE;
  payload: {
    laneId: string;
    changes: Partial<LaneModel>,
  };
}

export function createLane(title: string): CreateLaneAction {
  return {
    type: LaneActionTypes.LANE_CREATE,
    payload: {
      lane: {
        id: uuid.v4(),
        title,
        cards: [],
      },
    },
  };
}

export function deleteLane(laneId: string): DeleteLaneAction {
  return {
    type: LaneActionTypes.LANE_DELETE,
    payload: {
      laneId,
    },
  };
}

export function updateLane(laneId: string, changes: Partial<LaneModel>): UpdateLaneAction {
  return {
    type: LaneActionTypes.LANE_UPDATE,
    payload: {
      laneId,
      changes,
    },
  };
}
