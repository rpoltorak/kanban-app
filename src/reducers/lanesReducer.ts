import { CardActionTypes } from "../actions/cardActions";
import { LaneActionTypes } from "../actions/laneActions";
import { RootAction } from "./rootReducer";
import LaneModel from "../models/laneModel";
import CardModel from "../models/cardModel";

export interface LanesState {
  byId: {
    [laneId: string]: LaneModel;
  };
  allIds: string[];
}

const initialState = {
  byId: {},
  allIds: [],
};

export default (state: LanesState = initialState, action: RootAction) => {
  switch (action.type) {
    case LaneActionTypes.LANE_CREATE:
      return createLane(state, action.payload.lane);

    case LaneActionTypes.LANE_DELETE:
      return deleteLane(state, action.payload.laneId);

    case LaneActionTypes.LANE_UPDATE: {
      const { laneId, changes } = action.payload;
      return updateLane(state, laneId, changes);
    }

    case CardActionTypes.CARD_CREATE:
      return createCard(state, action.payload.card);

    case CardActionTypes.CARD_DELETE:
      return deleteCard(state, action.payload);

    default: return state;
  }
};

function createLane(state: LanesState, lane: LaneModel): LanesState {
  return {
    byId: {
      ...state.byId,
      [lane.id]: lane,
    },
    allIds: [...state.allIds, lane.id],
  };
}

function deleteLane(state: LanesState, laneId: string): LanesState {
  const {[laneId]: deletedLane, ...otherLanes} = state.byId;

  return {
    byId: otherLanes,
    allIds: state.allIds.filter((id: string): boolean => id !== laneId),
  };
}

function updateLane(state: LanesState, laneId: string, changes: Partial<LaneModel>): LanesState {
  const lane = state.byId[laneId];

  return {
    byId: {
      ...state.byId,
      [laneId]: {
        ...lane,
        ...changes,
      },
    },
    allIds: state.allIds,
  };
}

function createCard(state: LanesState, card: CardModel): LanesState {
  const lane = state.byId[card.laneId];

  return {
    byId: {
      ...state.byId,
      [card.laneId]: {
        ...lane,
        cards: [...lane.cards, card.id],
      },
    },
    allIds: state.allIds,
  };
}

function deleteCard(state: LanesState, params: { cardId: string, laneId: string }): LanesState {
  const lane = state.byId[params.laneId];

  return {
    byId: {
      ...state.byId,
      [params.laneId]: {
        ...lane,
        cards: lane.cards.filter((cardId: string) => cardId !== params.cardId),
      },
    },
    allIds: state.allIds,
  };
}
