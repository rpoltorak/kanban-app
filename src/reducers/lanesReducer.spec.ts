
import lanesReducer from "./lanesReducer";
import LaneModel from "../models/laneModel";
import { createLane, deleteLane, updateLane } from "../actions/laneActions";
import CardModel from "../models/cardModel";
import { createCard, deleteCard } from "../actions/cardActions";

const lane: LaneModel = { id: "1", title: "current title", cards: ["21", "22"]  };
const otherLane: LaneModel = { id: "2", title: "current title", cards: [] };
const updatedLane: LaneModel = { id: "1", title: "updated title", cards: ["21", "22"] };

const initialState = lanesReducer(undefined, {} as any);

describe("Lanes Reducer", () => {

  describe("snapshot", () => {
    it("should match a snapshot", () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe("lane_create", () => {
    it("should add a new lane as the first element", () => {
      const action = createLane("title");
      const state = lanesReducer(initialState, action);
      const { id } = action.payload.lane;

      expect(state.byId[id]).toEqual(action.payload.lane);
      expect(state.allIds).toHaveLength(1);
      expect(state.allIds[0]).toEqual(id);
    });
  });

  describe("lane_delete", () => {
    it("should delete existing lane with specified id", () => {
      const prevState = {
        ...initialState,
        byId: {
          [lane.id]: lane,
        },
        allIds: [lane.id],
      };

      const action = deleteLane(lane.id);
      const nextState = lanesReducer(prevState, action);

      expect(nextState.allIds).toHaveLength(0);
      expect(nextState.byId[lane.id]).toBeUndefined();
    });
  });

  describe("lane_update", () => {
    it("should update existing lane's title and description", () => {
      const prevState = {
        ...initialState,
        byId: {
          [lane.id]: lane,
        },
        allIds: [lane.id],
      };

      const action = updateLane(lane.id, { title: "updated title" });
      const nextState = lanesReducer(prevState, action);

      expect(nextState.byId[lane.id]).toEqual(updatedLane);
    });
  });

  describe("card_create", () => {
    it("should add cardId to related lane", () => {
      const prevState = {
        ...initialState,
        byId: {
          [lane.id]: lane,
        },
        allIds: [lane.id],
      };

      const action = createCard("title", "description", lane.id);
      const nextState = lanesReducer(prevState, action);

      expect(nextState.byId[lane.id].cards[lane.cards.length]).toEqual(action.payload.card.id);
      expect(nextState.byId[lane.id].cards).toHaveLength(lane.cards.length + 1);
    });
  });

  describe("card_delete", () => {
    it("should delete card related to specified lane", () => {
      const prevState = {
        ...initialState,
        byId: {
          [lane.id]: lane,
        },
        allIds: [lane.id],
      };

      const action = deleteCard(lane.cards[0], lane.id);
      const nextState = lanesReducer(prevState, action);

      expect(nextState.byId[lane.id].cards).toHaveLength(lane.cards.length - 1);
    });
  });
});
