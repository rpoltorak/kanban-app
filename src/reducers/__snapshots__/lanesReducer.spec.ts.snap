// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lanes Reducer snapshot should match a snapshot 1`] = `
Object {
  "allIds": Array [],
  "byId": Object {},
}
`;
